scripts/salt-setup.sh

```diff
@@ -46,6 +46,53 @@ function apply_optional_states {
   fi
 }
 
+# This adds entities to the 'roles' key to be used during actual cluster install. Hence, this does not use the temporary config location.
+function add_single_role_for_cluster_salt {
+  local role=${1}
+  ${SALT_PATH}/bin/salt-call --local grains.append roles ${role} -l info --log-file=/tmp/salt-build-${saltenv}.log --log-file-level=debug
```

Comment:
Grains are stored in the filesystem somewhere /etc/salt-master/ so it's easier just to create the file.

Reply:
These get stored in /etc/salt/grains, which is where the grains set up dynamically by CB are also stored.

Would prefer to leave it via the salt invocation, so as not to worry about formats, existing contents of the file - and allow for future additions without worrying about formatting. Let salt take care of all of that.